### PR TITLE
Max/create

### PR DESCRIPTION
**Pull request by Bettelstab**

### CODE REVIEW

**Comment by Bettelstab** [comment]
I'm resolving typing issues...

File: src/common.css [hunk 1]
```diff
@@ -12,13 +12,125 @@
...
-}
\ No newline at end of file
+}
+
+.loading-animation {
+    opacity: 1;
+    background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64" width="64" height="64"><style>@keyframes dash-animation { to { transform: rotate(240deg) translateZ(0); } } %23circle { animation: 3s dash-animation infinite linear; transform: rotate(-120deg) translateZ(0); transform-origin: center; }</style><defs><clipPath id="hexClip"><path clip-rule="evenodd" d="M16 4.29h32L64 32 48 59.71H16L0 32zm4.62 8h22.76L54.76 32 43.38 51.71H20.62L9.24 32z"/></clipPath></defs><path fill-rule="evenodd" d="M16 4.29h32L64 32 48 59.71H16L0 32zm4.62 8h22.76L54.76 32 43.38 51.71H20.62L9.24 32z" fill="%23fff" opacity=".2"/><g clip-path="url(%23hexClip)"><circle id="circle" cx="32" cy="32" r="16" fill="none" stroke-width="32" stroke="%23F6AE2D" stroke-dasharray="16.666 84.666"/></g></svg>');
+    background-repeat: no-repeat;
+    background-position: center;
+    background-size: 72px;
```
Comment (sisou) [inline src/common.css]: Everything in REM? The whole `common.css`?

File: src/common.css [hunk 2]
```diff
@@ -12,13 +12,125 @@
...
+.info-box .info-text {
+    text-align: left;
+}
+
+.spacing-top-16 {
+    margin-top: 16px;
+}
+
+/* make code blocks look like keys on the keyboard */
+code {
```
Comment (sisou) [inline src/common.css]: Maybe we should use the `<kbd>` tag for this?

File: src/components/Identicon.js [hunk 3]
```diff
@@ -32,13 +26,28 @@ class Identicon { // eslint-disable-line no-unused-vars
         this._updateIqon();
     }
 
+    /**
+     * @param {HTMLDivElement} [$el]
+     * @returns {HTMLDivElement}
+     */
+    _createElement($el) {
+        const $element = $el || document.createElement('div');
```
Comment (sisou) [inline src/components/Identicon.js]: :+1:

File: src/components/Identicon.js [hunk 4]
```diff
@@ -1,27 +1,21 @@
...
-     * @param {HTMLImageElement} [el]
+     * @param {HTMLDivElement} [$el]
      */
-    constructor(address, el) {
+    constructor(address, $el) {
         this._address = address;
 
-        /** @type {HTMLImageElement} */
-        this.$el = el || this.getElement();
+        this.$el = this._createElement($el);
```
Comment (sisou) [inline src/components/Identicon.js]: You made the `_createElement` method static below in `PassphraseInput`, but not here. We should have a standard.

File: src/components/RecoveryWords.css [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    margin-bottom: 24px;
+}
+
+.recovery-words .info-box {
+    margin-bottom: 24px;
+}
+
+.recovery-words .words-container {
+    width: 100%;
+    max-width: 360px;
```
Comment (sisou) [inline src/components/RecoveryWords.css]: Everything in `rem`?

File: src/components/RecoveryWords.css [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+.recovery-words .words-container .word-section:hover .word-content{
+    display: initial;
+}
+
+.recovery-words .words-container .word-section:hover .word-placeholder{
+    display: none;
+}
+
+.recovery-words .words-container:before {
+    content: 'Recovery Words';
```
Comment (sisou) [inline src/components/RecoveryWords.css]: Not translatable, I think?

File: src/components/RecoveryWords.css [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    background: url('data:image/svg+xml;base64,PHN2ZyBmaWxsPSIjZmZmIiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNMCAwaDI0djI0SDB6IiBmaWxsPSJub25lIi8+PHBhdGggZD0iTTE4IDhoLTFWNmMwLTIuNzYtMi4yNC01LTUtNVM3IDMuMjQgNyA2djJINmMtMS4xIDAtMiAuOS0yIDJ2MTBjMCAxLjEuOSAyIDIgMmgxMmMxLjEgMCAyLS45IDItMlYxMGMwLTEuMS0uOS0yLTItMnptLTYgOWMtMS4xIDAtMi0uOS0yLTJzLjktMiAyLTIgMiAuOSAyIDItLjkgMi0yIDJ6bTMuMS05SDguOVY2YzAtMS43MSAxLjM5LTMuMSAzLjEtMy4xIDEuNzEgMCAzLjEgMS4zOSAzLjEgMy4xdjJ6Ii8+PC9zdmc+');
+    background-size: 14px;
+    background-position: 9px center;
+    background-repeat: no-repeat;
+    background-color: black;
+    padding-left: 16px;
+    transition: opacity 300ms;
+}
+
+.recovery-words .words-container.x-entry .x-word {
```
Comment (sisou) [inline src/components/RecoveryWords.css]: `x-*` classes do not exist anymore.

File: src/components/RecoveryWords.css [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    animation: x-recovery-phrase 300ms;
```
Comment (sisou) [inline src/components/RecoveryWords.css]: Animation does not exist.

File: src/components/RecoveryWords.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * @param {HTMLElement} [$el]
+     * @returns {Element}
+     * */
+    _createElement($el) {
+        $el = $el || document.createElement('div');
+        $el.classList.add('recovery-words');
+
+        $el.innerHTML = `
+            <h1>Backup your 24 Recovery Words</h1>
```
Comment (sisou) [inline src/components/RecoveryWords.js]: Missing i18n attribute.

File: src/components/RecoveryWords.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+            <h2 secondary>
```
Comment (sisou) [inline src/components/RecoveryWords.js]: Missing i18n attribute.

File: src/components/RecoveryWords.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+                Write down and physically store the complete following list of 24 Account Recovery Words
+                at a <strong>SAFE and SECRET</strong> place to recover this account in the future.
+            </h2>
+            <div class="grow"></div>
+            <div class="words-container"></div>
+            <div class="info-box">
+                <i class="info-icon"></i>
+                <p class="info-text">Move your mouse over the numbers or tap them to reveal each word.</p>
```
Comment (sisou) [inline src/components/RecoveryWords.js]: Missing i18n attribute.

File: src/components/RecoveryWords.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+            </div>
+            <div class="spacing-bottom center warning">
+                <strong>Anyone with access to these words can steal all your funds!</strong>
```
Comment (sisou) [inline src/components/RecoveryWords.js]: Missing i18n attribute.

File: src/components/RecoveryWords.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+            </div>
+            <x-grow></x-grow>
```
Comment (sisou) [inline src/components/RecoveryWords.js]: Rename `x-grow`, or better: make it a CSS class.

File: src/components/ValidateWords.css [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    color: var(--secondary-text-color);
+}
+
+#validate-words button.wrong {
+    background-color: var(--error-color);
+    border-color: var(--error-color);
+    color: var(--secondary-cta-color);
+}
+
+x-wordlist {
```
Comment (sisou) [inline src/components/ValidateWords.css]: Rename to other than `x-*`, if still needed.

File: src/lib/I18n.js [hunk 15]
```diff
@@ -51,7 +51,13 @@ class I18n { // eslint-disable-line no-unused-vars
             translateElements(`i18n-${tag}`, (element, translation) => element.setAttribute(tag, translation));
         };
 
-        translateElements('i18n', (element, translation) => { element.textContent = translation; });
+        translateElements('i18n', (element, translation) => {
+            const doc = this.parser.parseFromString(translation, 'text/html');
+            const noHtml = /** @type {string} */ (doc.body.textContent);
+            let withMarkup = noHtml.replace('[strong]', '<strong>');
+            withMarkup = withMarkup.replace('[/strong]', '</strong>');
```
Comment (sisou) [inline src/lib/I18n.js]: You can chain the `replace()` calls directly in one line.

File: src/nimiq-style.css [hunk 16]
```diff
@@ -458,7 +458,7 @@ h1,
     padding: 0;
 }
 
-h1+h2, h2[secondary] {
+h1+h2, h2.secondary {
```
Comment (sisou) [inline src/nimiq-style.css]: Missing spaces around the plus sign.

File: src/nimiq-style.css [hunk 17]
```diff
@@ -507,30 +507,33 @@ button[secondary] {
...
-.nimiq-dark button[secondary] {
+.nimiq-dark a.secondary,
+.nimiq-dark button.secondary {
     color: #5c5c5c;
 }
 
-[secondary]+button,
-[secondary]+[button] {
+.secondary+button,
+.secondary+[button] {
```
Comment (sisou) [inline src/nimiq-style.css]: Please add spaces around the plus signs.

File: src/nimiq-style.css [hunk 18]
```diff
@@ -507,30 +507,33 @@ button[secondary] {
...
     margin-top: 5rem;
 }
 
-[secondary]+[secondary],
-button+[secondary],
-[button]+[secondary] {
+.secondary+.secondary,
+button+.secondary,
+.button+.secondary {
```
Comment (sisou) [inline src/nimiq-style.css]: Please add spaces around the plus signs.

File: src/nimiq-style.css [hunk 19]
```diff
@@ -542,5 +545,4 @@ li {
         font-size: 1.625rem;
         padding-bottom: 1rem;
     }
-}
-
+}
```
Comment (sisou) [inline src/nimiq-style.css]: Missing newline.

File: src/request/create/ChooseIdenticon.css [hunk 20]
```diff
@@ -0,0 +1,191 @@
+
+
+#choose-identicon {
```
Comment (sisou) [inline src/request/create/ChooseIdenticon.css]: Remove empty leading lines.

File: src/request/create/ChooseIdenticon.js [hunk 21]
```diff
@@ -0,0 +1,127 @@
+class ChooseIdenticon extends Nimiq.Observable {
+    /**
+     * @param {EncryptionType} encryptionType
+     * @param {HTMLElement} [$el]
+     */
+    constructor(encryptionType, $el) {
```
Comment (sisou) [inline src/request/create/ChooseIdenticon.js]: I understand that the encryption type is necessary for instantiating a `Key`, but it's rather unneeded in the identicon selector, no? Can't the selector return a `KeyPair`, instead of a `Key` instance?

File: src/request/create/ChooseIdenticon.js [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+        if (this.$selectedIdenticon) {
+            this.$selectedIdenticon.classList.add('returning');
+            this.$selectedIdenticon.classList.remove('selected');
+        }
+
+        this.$el.classList.remove('selected');
+    }
+}
+
+ChooseIdenticon.EVENTS = {
```
Comment (sisou) [inline src/request/create/ChooseIdenticon.js]: Should be named `ChooseIdenticons.Events`, same as in other components.

File: src/request/create/CreateHigh.js [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+            window.location.hash = CreateHigh.Pages.VALIDATE_WORDS;
+        });
+
+        this._validateWords.on(ValidateWords.Events.BACK, () => {
+            window.location.hash = CreateHigh.Pages.RECOVERY_WORDS;
+        });
+
+        this._validateWords.on(ValidateWords.Events.BACK, () => {
+            window.location.hash = CreateHigh.Pages.RECOVERY_WORDS;
+        });
```
Comment (sisou) [inline src/request/create/CreateHigh.js]: This is double. It's already defined 5 lines up.

File: src/request/create/CreateHigh.js [hunk 24]
```diff
@@ -0,0 +1,113 @@
...
+
+        this._validateWords.on(ValidateWords.Events.VALIDATED, () => {
+            window.location.hash = CreateHigh.Pages.SET_PASSPHRASE;
+        });
+
+        this._passphraseSetter.on(
+            PassphraseInput.Events.PASSPHRASE_ENTERED,
+            /** @param {string} passphrase */passphrase => {
```
Comment (sisou) [inline src/request/create/CreateHigh.js]: Missing space after `*/`.

File: src/request/create/CreateHigh.js [hunk 25]
```diff
@@ -0,0 +1,113 @@
...
+                }
+            },
+        );
+    }
+
+    run() {
+        // go to start page
+        window.location.hash = CreateHigh.Pages.CHOOSE_IDENTICON;
+
+        this._chooseIdenticon.generateIdenticons();
```
Comment (sisou) [inline src/request/create/CreateHigh.js]: Maybe switch the order of these two statements so that the identicons will be generated before the route is set?

File: src/request/create/CreateLow.js [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+            ChooseIdenticon.EVENTS.CHOOSE_IDENTICON,
+            /** @param {Key} key */
+            key => {
+                this._selectedKey = key;
+                window.location.hash = CreateLow.Pages.SET_PIN;
+                this._pinInput.open();
+            },
+        );
+
+        this._pinInput.on(PinInput.Events.PIN_ENTERED, /** @param {number} pin */ async pin => {
```
Comment (sisou) [inline src/request/create/CreateLow.js]: I liked this above, when it was split into two or three lines for readability.

File: src/request/create/CreateLow.js [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+                this._resolve(await KeyStore.instance.put(this._selectedKey, pin.toString()));
+            }
+        });
+    }
+
+    run() {
+        // go to start page
+        window.location.hash = CreateLow.Pages.CHOOSE_IDENTICON;
+
+        this._chooseIdenticon.generateIdenticons();
```
Comment (sisou) [inline src/request/create/CreateLow.js]: Same as above, maybe switch order of the two statements.

File: src/request/create/index.html [hunk 28]
```diff
@@ -0,0 +1,127 @@
...
+    <script defer src="CreateApi.js"></script>
+    <script defer src="index.js"></script>
+
+    <link href="../../nimiq-style.css" rel="stylesheet">
+    <link rel="stylesheet" href="../../common.css">
+    <link rel="stylesheet" href="../../components/PinInput.css">
+    <link rel="stylesheet" href="../../components/PassphraseInput.css">
+    <link rel="stylesheet" href="../../components/RecoveryWords.css">
+    <link rel="stylesheet" href="../../components/ValidateWords.css">
+    <link rel="stylesheet" href="./privacy-agent.css">
```
Comment (sisou) [inline src/request/create/index.html]: The PrivacyAgent is needed in other requests, too. It should be made a "component".

File: src/request/create/index.html [hunk 29]
```diff
@@ -0,0 +1,127 @@
...
+</head>
+<body>
+<div id="app">
+
+    <div id="choose-identicon" class="page">
+
+    </div>
+
+    <div id="privacy-agent" class="page">
+        <h1>Backup your Account</h1>
```
Comment (sisou) [inline src/request/create/index.html]: Translation missing for this whole file.

File: src/request/create/privacy-agent.css [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+    font-weight: 500;
+    font-size: 24px;
+}
+
+.privacy-agent p {
+    max-width: 400px;
+    font-size: 15px;
+}
+
+.privacy-agent svg{
```
Comment (sisou) [inline src/request/create/privacy-agent.css]: Missing space before `{`.

File: src/request/create/privacy-agent.css [hunk 31]
```diff
@@ -0,0 +1,52 @@
...
+    opacity: 0.8;
+    display: block;
+}
+
+.privacy-agent button{
```
Comment (sisou) [inline src/request/create/privacy-agent.css]: Missing space before `{`.

File: src/request/sign-transaction/SignTransaction.js [hunk 32]
```diff
@@ -1,4 +1,4 @@
-class SignTransactionView extends Nimiq.Observable { // eslint-disable-line no-unused-vars
```
Comment (sisou) [inline src/request/sign-transaction/SignTransaction.js]: Why are you even changing `sign-transaction` files in a PR about the `create` request? Should be moved out into a separate PR.

File: src/request/sign-transaction/SignTransaction.js [hunk 33]
```diff
@@ -1,4 +1,4 @@
...
+class SignTransaction extends Nimiq.Observable { // eslint-disable-line no-unused-vars
```
Comment (sisou) [inline src/request/sign-transaction/SignTransaction.js]: Does not need to extend `Nimiq.Observable` anymore if it doesn't use `fire()`.

File: src/translations/index.js [hunk 34]
```diff
@@ -7,6 +7,11 @@ const TRANSLATIONS = {
         'passphrase-placeholder': 'Enter passphrase',
         'sign-tx-wrong-passphrase': 'Wrong passphrase, please try again',
         'sign-tx-button-enter-pin': 'Enter PIN',
+        'create-choose-identicon-header1': 'Choose Your Account Avatar',
+        'create-set-passphrase-header1': 'Set a Pass Phrase',
+        'create-set-passphrase-header2': 'Please enter a Pass Phrase to secure your account.',
+        'create-set-passphrase-warning': `The Pass Phrase is [strong]not[/strong] an alternative for your 24 Recovery
```
Comment (sisou) [inline src/translations/index.js]: `Pass Phrase` should be `Passphrase`.

File: package-lock.json [hunk 35]
```diff
@@ -16,7 +16,7 @@
       "integrity": "sha1-63d99gEXI6OxTopywIBcjoZ0a9I=",
       "dev": true,
       "requires": {
-        "mime-types": "2.1.18",
+        "mime-types": "~2.1.18",
```
[the remark this answers is not in the document]
Comment (sisou) [reply package-lock.json]: I think the problem lays in your tooling, because when I execute `npm i`, it does _not_ add these symbols. There is no reason why your PR should update this file.

File: src/translations/index.js [hunk 36]
```diff
@@ -5,11 +5,11 @@ const TRANSLATIONS = { // eslint-disable-line no-unused-vars
...
         'passphrase-confirm': 'Confirm',
         'passphrase-placeholder': 'Enter passphrase',
-        'sign-tx-wrong-passphrase': 'Wrong passphrase, please try again',
+        'sign-tx-wrong-passphrase': 'Wrong Passphrase, please try again',
         'sign-tx-button-enter-pin': 'Enter PIN',
         'create-choose-identicon-header1': 'Choose Your Account Avatar',
-        'create-set-passphrase-header1': 'Set a Pass Phrase',
-        'create-set-passphrase-header2': 'Please enter a Pass Phrase to secure your account.',
+        'create-set-passphrase-header1': 'Set a Passphrase',
+        'create-set-passphrase-header2': 'Please enter a Passpt hrase to secure your account.',
```
Comment (sisou) [inline src/translations/index.js]: Typo.

File: src/request/create/CreateHigh.js (hunk 25 above)
Comment (sisou) [reply src/request/create/CreateHigh.js]: Order is on purpose to enable identicon spread animation.

File: src/request/create/CreateLow.js (hunk 27 above)
Comment (sisou) [reply src/request/create/CreateLow.js]: Order is on purpose to enable identicon spread animation.

File: src/request/create/index.html (hunk 28 above)
Comment (sisou) [reply src/request/create/index.html]: To be done in svub's import request PR.

File: src/components/PassphraseInput.js [hunk 37]
```diff
@@ -27,13 +27,15 @@ class PassphraseInput extends Nimiq.Observable {
...
 
     /**
+     * @param {HTMLFormElement} [$el]
      * @returns {HTMLFormElement}
      */
-    _createElement() {
-        const el = document.createElement('form');
-        el.classList.add('passphrase-input');
+    static _createElement($el) {
+        $el = $el || document.createElement('form');
```
Comment (danimoh) [inline src/components/PassphraseInput.js]: if $el was provided as an argument, should check that it is a form element

File: src/request/create/ChooseIdenticon.js [hunk 38]
```diff
@@ -0,0 +1,128 @@
...
+    _onIdenticonSelected($identicon, address) {
+        const $returningIdenticon = this.$el.querySelector('.identicon.returning');
+        if ($returningIdenticon) {
+            $returningIdenticon.classList.remove('returning');
+        }
+
+        this.$confirmButton.addEventListener('click', () => this.fire(
+            ChooseIdenticon.Events.CHOOSE_IDENTICON,
+            this._volatileKeys[address],
+        ));
```
Comment (danimoh) [inline src/request/create/ChooseIdenticon.js]: this click listener does not get revoked when the selected identicon is dismissed.

File: src/request/create/CreateHigh.js [hunk 39]
```diff
@@ -0,0 +1,114 @@
...
+     * @param {Function} resolve
+     * @param {Function} reject
+     */
+    constructor(request, resolve, reject) {
+        this._resolve = resolve;
+        this._reject = reject;
+
+        // set html elements
+        /** @type {HTMLDivElement} */
+        this.$chooseIdenticon = (document.getElementById(CreateHigh.Pages.CHOOSE_IDENTICON));
```
Comment (danimoh) [inline src/request/create/CreateHigh.js]: surrounding parenthesis are not necessary

File: src/request/create/CreateHigh.js [hunk 40]
```diff
@@ -0,0 +1,114 @@
...
+        // wire up logic
+        this._chooseIdenticon.on(
+            ChooseIdenticon.Events.CHOOSE_IDENTICON,
+            /** @param {Nimiq.KeyPair} keyPair */
+            keyPair => {
+                this._selectedKeyPair = keyPair;
+                const keyAsUInt8 = keyPair.privateKey.serialize();
+                this._recoveryWords.privateKey = keyAsUInt8;
+                this._validateWords.privateKey = keyAsUInt8;
+                this._validateWords.reset();
```
Comment (danimoh) [inline src/request/create/CreateHigh.js]: let the private key setter reset the validate words component (see above)

File: src/request/create/CreateHigh.js [hunk 41]
```diff
@@ -0,0 +1,114 @@
...
+                $passphraseSetter.classList.add('display-none');
+            },
+        );
+
+        this._passphraseGetter.on(
+            PassphraseInput.Events.PASSPHRASE_ENTERED,
+            /** @param {string} passphrase */ async passphrase => {
+                if (this._passphrase !== passphrase) {
+                    this._passphraseGetter.onPassphraseIncorrect();
+                    this._passphraseGetter.reset();
```
Comment (danimoh) [inline src/request/create/CreateHigh.js]: `onPassphraseIncorrect` already resets the passphrase

File: src/request/create/CreateHigh.js [hunk 42]
```diff
@@ -0,0 +1,114 @@
...
+                    $passphraseConfirm.classList.add('display-none');
+                    $passphraseSetter.classList.remove('display-none');
```
Comment (danimoh) [inline src/request/create/CreateHigh.js]: I don't think it's good ux to go back to the password setter.
The user should get some message.
Also, `onPassphraseIncorrect` triggers a shake animation that you're not waiting for

File: src/request/create/CreateLow.js [hunk 43]
```diff
@@ -0,0 +1,80 @@
...
+     * @param {Function} resolve
+     * @param {Function} reject
+     */
+    constructor(request, resolve, reject) {
+        this._resolve = resolve;
+        this._reject = reject;
+
+        // set html elements
+        /** @type {HTMLDivElement} */
+        this.$chooseIdenticon = (document.getElementById(CreateLow.Pages.CHOOSE_IDENTICON));
```
Comment (danimoh) [inline src/request/create/CreateLow.js]: parentheses unnecessary

File: src/request/create/CreateLow.js [hunk 44]
```diff
@@ -0,0 +1,80 @@
...
+
+        /** @type {HTMLDivElement} */
+        this.$setPin = (document.getElementById(CreateLow.Pages.SET_PIN));
+
+        /** @type {HTMLDivElement} */
+        const $pinInput = (this.$setPin.querySelector('.pinpad'));
+
+        /** @type {HTMLDivElement} */
+        const $confirmMessage = /** @type {HTMLDivElement} */ (this.$setPin.querySelector('.confirm-message'));
```
Comment (danimoh) [inline src/request/create/CreateLow.js]: unnecessary double type declaration

File: src/request/create/CreateLow.js [hunk 45]
```diff
@@ -0,0 +1,80 @@
...
+
+        /** @type {HTMLDivElement} */
+        const $notMatchingMessage = /** @type {HTMLDivElement} */ (this.$setPin.querySelector('.not-matching-message'));
```
Comment (danimoh) [inline src/request/create/CreateLow.js]: unnecessary double type declaration

File: src/request/create/CreateLow.js [hunk 46]
```diff
@@ -0,0 +1,80 @@
...
+                $notMatchingMessage.classList.add('hidden');
+
+                if (!this._pin) {
+                    this._pin = pin;
+                    this._pinInput.reset();
+                    $confirmMessage.classList.remove('hidden');
+                } else if (this._pin !== pin) {
+                    this._pinInput.onPinIncorrect();
+                    this._pin = null;
+                    $notMatchingMessage.classList.remove('hidden');
```
Comment (danimoh) [inline src/request/create/CreateLow.js]: good ux that the user has to redefine his pin ?

File: src/request/create/index.html [hunk 47]
```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="Content-Security-Policy"
+          content="default-src 'self' 'unsafe-eval' https://cdn.nimiq.com;
```
Comment (danimoh) [inline src/request/create/index.html]: does that mean that `unsafe-eval` is allowed?

File: src/request/sign-transaction/SignTransactionWithPassphrase.js [hunk 48]
```diff
@@ -1,32 +1,41 @@
...
+    constructor(txRequest, resolve, reject) {
         super();
-
         this._txRequest = txRequest;
+        this._resolve = resolve;
+        this._reject = reject;
+
+        // set html elements
+        /** @type {HTMLDivElement} */
+        this.$rootElement = (document.getElementById('app'));
```
Comment (danimoh) [inline src/request/sign-transaction/SignTransactionWithPassphrase.js]: no need for the parentheses

File: src/request/create/CreateLow.js [hunk 49]
```diff
@@ -0,0 +1,80 @@
...
+                } else {
+                    document.body.classList.add('loading');
+                    const key = new Key(this._selectedKeyPair, request.type);
```
Comment (danimoh) [inline src/request/create/CreateLow.js]: this._pinInput.close();

File: src/lib/I18n.js [hunk 50]
```diff
@@ -53,7 +53,12 @@ class I18n { // eslint-disable-line no-unused-vars
             translateElements(`i18n-${tag}`, (element, translation) => element.setAttribute(tag, translation));
         };
 
-        translateElements('i18n', (element, translation) => { element.textContent = translation; });
+        translateElements('i18n', (element, translation) => {
+            const doc = this.parser.parseFromString(translation, 'text/html');
+            const noHtml = /** @type {string} */ (doc.body.textContent);
+            const withMarkup = noHtml.replace('[strong]', '<strong>').replace('[/strong]', '</strong>');
+            element.innerHTML = withMarkup;
+        });
```
[the remark this answers is not in the document]
Comment (sisou) [reply src/lib/I18n.js]: The translations are parsed as HTML and then taken as `textContent` in lines 57 and 58, thus translations are cleaned of all tags. Then only certain tags are replaced, like `strong` here.

File: src/request/create/ChooseIdenticon.js (hunk 38 above)
Comment (sisou) [reply src/request/create/ChooseIdenticon.js]: You mean when generating new identicons?

File: src/request/create/CreateHigh.js (hunk 39 above)
Comment (sisou) [reply src/request/create/CreateHigh.js]: Yes they are, as otherwise TypeScript does not apply the @type annotation. Even when it's two lines like this.

File: src/request/create/index.html (hunk 47 above)
Comment (sisou) [reply src/request/create/index.html]: There was something about it being required for WebAssembly to work under iOS, IIRC.

Comment (Bettelstab) [reply src/request/create/index.html]: More or less. Core needs it to determine if it has to use it's babel-built fallback version.
